### ISSUE TITLE
refactor: Consistent use of `walletName` vs `walletFileName`

### ISSUE THREAD

**Issue by theborakompanioni** [joinmarket-webui/jam#640]
It is not totally clear, when a wallet name is needed, if the filename (including extension `.jmdat`) or the "display name" (without extension) is needed. Hence, everytime the wallet name is rendered, it is first given to util function `displayWalletName`. This can be made more obvious and the wallet filename, as well as the wallet "display name" should be made statically available in the `WalletContext`.

Hint: There is already a type `Api.WalletName` which  actually means the "wallet filename". Change/introduce types to make this more obvious.

**Comment by theborakompanioni** [comment]
Blocked by #621

**Comment by theborakompanioni** [comment]
Idea: When this is consistently done, Jam could store meta information in the filename, e.g. wallet birthday (`jam_<wallet_name>_20090301.jmdat`). This might be one way for Jam to store some info about the wallet permanently.

**Comment by kristapsk** [comment]
> Idea: When this is consistently done, Jam could store meta information in the filename, e.g. wallet birthday (`jam_<wallet_name>_20090103.jmdat`). This might be one way for Jam to store some info about the wallet permanently.

Various metadata could be stored in wallet .jmdat file itself by JM code.

**Comment by theborakompanioni** [comment]
> > Idea: When this is consistently done, Jam could store meta information in the filename, e.g. wallet birthday (`jam_<wallet_name>_20090103.jmdat`). This might be one way for Jam to store some info about the wallet permanently.
> 
> Various metadata could be stored in wallet .jmdat file itself by JM code.

Thanks @kristapsk!
It was a very stupid thought of mine. :grimacing: 

Let's focus on more important things first!
(e.g. access to `blocks` (blockheight) would be a nice to have for Jam - will open a ticket).
